pkg/networkserver/mac/ping_slot_channel.go

```diff
@@ -94,15 +98,21 @@ func HandlePingSlotChannelAns(ctx context.Context, dev *ttnpb.EndDevice, pld *tt
 		ttnpb.MACCommandIdentifier_CID_PING_SLOT_CHANNEL,
 		false,
 		func(cmd *ttnpb.MACCommand) error {
+			if !pld.DataRateIndexAck || !pld.FrequencyAck {
```

Comment:
Nack any means nack of the command yeah?

Reply:
<img width="811" alt="image" src="https://github.com/TheThingsNetwork/lorawan-stack/assets/36161392/9a3dd8f7-bd44-4aff-b305-4428dde6f2a3">
Correct, and almost all MAC commands have this behavior. There is only one outlier in the 1.1 specification, `RejoinParamSetupReq`, which can partially fail:
<img width="846" alt="image" src="https://github.com/TheThingsNetwork/lorawan-stack/assets/36161392/a14d10ff-0297-4906-a607-84e68f2b4620">